README.md

```diff
@@ -20,110 +13,1177 @@ visit https://pkg.go.dev/github.com/pinecone-io/go-pinecone/pinecone.
+
+    res, err := idxConnection.QueryByVectorValues(ctx, &pinecone.QueryByVectorValuesRequest{
+        Vector:        queryVector,
+        TopK:          3,
+		Filter: 	   metadataFilter,
+        IncludeValues: true,
+    })
+    if err != nil {
+        log.Fatalf("Error encountered when querying by vector: %v", err)
+    } else {
+        fmt.Printf(prettifyStruct(res))
+    }
+}
+
+// Returns:
+// {
+//   "matches": [
+//     {
+//       "vector": {
+//         "id": "B",
+//         "values": [
+//           0.2,
+//           0.2,
+//           0.2,
+//           0.2,
+//           0.2,
+//           0.2,
+//           0.2,
+//           0.2
+//         ]
+//       },
+//       "score": 1
+//     },
+//     {
+//       "vector": {
+//         "id": "C",
+//         "values": [
+//           0.3,
+//           0.3,
+//           0.3,
+//           0.3,
+//           0.3,
+//           0.3,
+//           0.3,
+//           0.3
+//         ]
+//       },
+//       "score": 1
+//     },
+//     {
+//       "vector": {
+//         "id": "A",
+//         "values": [
+//           0.1,
+//           0.1,
+//           0.1,
+//           0.1,
+//           0.1,
+//           0.1,
+//           0.1,
+//           0.1
+//         ]
+//       },
+//       "score": 1
+//     }
+//   ],
+//   "usage": {
+//     "read_units": 6
+//   }
+// }
+```
+
+#### Query by vector id
+
+The following example queries the index `example-index` with a vector id value.
+
+```go
+package main
+
+import (
+    "context"
+    "encoding/json"
+    "fmt"
+    "log"
+
+    "github.com/pinecone-io/go-pinecone/pinecone"
+)
+
+func prettifyStruct(obj interface{}) string {
+	bytes, _ := json.MarshalIndent(obj, "", "  ")
+	return string(bytes)
+}
+
+func main() {
+    ctx := context.Background()
+
+    pc, err := pinecone.NewClient(pinecone.NewClientParams{
+        ApiKey: "YOUR_API_KEY",
+    })
+    if err != nil {
+        log.Fatalf("Failed to create Client: %v", err)
+    }
+
+    idx, err := pc.DescribeIndex(ctx, "example-index")
+    if err != nil {
+        log.Fatalf("Failed to describe index \"%v\": %v", idx.Name, err)
+    }
+
+    idxConnection, err := pc.Index(pinecone.NewIndexConnParams{Host: idx.Host, Namespace: "example-namespace"})
+    if err != nil {
+        log.Fatalf("Failed to create IndexConnection for Host %v: %v", idx.Host, err)
+    }
+
+	vectorId := "vector-id"
+    res, err := idxConnection.QueryByVectorId(ctx, &pinecone.&pinecone.QueryByVectorIdRequest{
+        VectorId:      vectorId,
+        TopK:          3,
+        IncludeValues: true,
+    })
+    if err != nil {
+        log.Fatalf("Error encountered when querying by vector ID `%v`: %v", vectorId, err)
+    } else {
+        fmt.Printf(prettifyStruct(res))
+    }
+}
+```
+
+### Delete vectors
+
+#### Delete vectors by ID
+
+The following example deletes a vector by its ID value from `example-index` and `example-namespace`. You can pass a slice of vector IDs to `DeleteVectorsById`.
+
```

Comment:
These examples need to have imports up top + be in a `main` function to be on par w/the earlier examples